clients/src/main/java/org/apache/kafka/clients/LeastLoadedNodeAlgorithm.java

```diff
@@ -24,15 +24,28 @@
 public enum LeastLoadedNodeAlgorithm {
   /**
    * default upstream kafka selection algorithm.
-   * attempts to minimize latency, but may resuult in stickiness
+   * attempts to minimize latency, but may result in stickiness
    * and hammering of dedicated controllers and brokers down
    * for maintenance
    */
   VANILLA,
   /**
-   * selects a random broker out of 3 candidates. candidates are preferrably
+   * selects a random broker out of 3 candidates. candidates are preferably
    * brokers with existing connections, but new connections will be initiated
    * to get the candidate pool up to 3.
    */
-  AT_LEAST_THREE
+  AT_LEAST_THREE,
+  /**
+   * selects a random broker out of all nodes except those for which the number
+   * of requests in flight is already at max.
+   */
+  RANDOM,
```

Comment:
could you add some comment here saying that this may end up with the client initializing connections to half of the brokers on average before the first MD request can be sentout.